app/services/global_search/common.rb

```diff
@@ -0,0 +1,23 @@
+module GlobalSearch
+
+  module Common
+
+    extend ActiveSupport::Concern
+
+    included do
+      attr_reader :facility, :query, :user
+    end
+
+    def initialize(user = nil, facility = nil, query = nil)
```

Comment:
Hmm, it feels a little strange to me to have an initializer in a concern. This is the kind of thing where I feel like inheritance might be better. If you named the class `GlobalSearch::Base`, it would work.